ui: Release notes signup - Email subscription form

koorosh opened the pull request:
Resolves: #43912

- Subscription form is responsible to accept user email, validate it and
submit an appropriate action.
- InputText component is styled according to design system.
- Current changes are not connected to any view or page (in total isolation) and can be safely merged without affecting existing functionality.

koorosh committed:
ui: Email subscription form

- Subscription form is responsible to accept user email, validate it and
submit an appropriate action.
- InputText component is styled according to design system.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/44744)
<!-- Reviewable:end -->

koorosh committed:
ui: Fix import statement and definition for Button component

Release note: None

koorosh committed:
ui: Extend Button and TextInput components

- Button component extended with styles for "primary" type
- InputText component is refactored to be as fully controlled
component.

Release note: None

koorosh committed:
ui: Correct validation message for invalid email address

- Display validation message for Input component with
absolute position to prevent layout to be expanded.
- Change validation message for invalid email address.

Release note: None

koorosh committed:
ui: Tests on EmailSubscriptionForm

Release note: None

koorosh committed:
ui: Validate input email on change

Add enhancements for InputText component:
- validation of input is performed during text change
- set fixed width for email input

Release note: None

dhartunian reviewed (approved):
@koorosh just a small nit

dhartunian commented:
bors r+

koorosh commented on pkg/ui/src/views/shared/components/emailSubscriptionForm/index.tsx:
```diff
@@ -0,0 +1,89 @@
+          type={"primary"}
+          onClick={this.handleSubmit}
+          disabled={!canSubmit}
+          className="email-subscription-form__submit-button"
```
Actually it requires min-width to avoid words wrapping in two lines (when window is resized to small width for example).
